### PR TITLE
fix(sql): use only added tables of connector

### PR DESCRIPTION
**Pull request by ArslanSaleem**
## Summary by CodeRabbit

- **New Features**
  - Added a method for checking allowed tables in SQL queries.
  - Enhanced security by introducing a new exception for malicious queries.

- **Bug Fixes**
  - Improved handling of table names in SQL queries to prevent potential misuse.

- **Tests**
  - Added new test methods to validate security checks on SQL queries and table name assignments.

### CODE REVIEW

**Comment by gventuri** [comment]
@ArslanSaleem could you fix the tests? I've just updated a error message based on CodeRabbit suggestion!

File: pandasai/helpers/code_manager.py [hunk 1]
```diff
@@ -326,6 +331,24 @@ def check_direct_sql_func_def_exists(self, node: ast.AST):
...
+                sql_query = node.value.value
+                table_names = extract_table_names(sql_query)
+                allowed_table_names = [df.table_name for df in self._dfs]
+                if any(
+                    table_name not in allowed_table_names for table_name in table_names
+                ):
+                    raise MaliciousQueryError(
+                        "Query uses unauthorized tables: {unauthorized_tables}. Please add them as new datatables or update the query."
+                    )
+
```
[the remark this answers is not in the document]
Comment (gventuri) [reply pandasai/helpers/code_manager.py:351]: @mspronesti good catch, I agree!

**Comment by gventuri** [comment]
Also, since the logic for the "direct sql" is becoming more complex, we should probably move it all outside of the code manager! But we can probably create a separate PR for this!

**Comment by ArslanSaleem** [comment]
> Also, since the logic for the "direct sql" is becoming more complex, we should probably move it all outside of the code manager! But we can probably create a separate PR for this!

@gventuri  Yes, absolutely and more scenarios will come

**Pull request merged**